source/common/filesystem/posix/filesystem_impl.cc

```diff
@@ -28,17 +28,38 @@ FileImplPosix::~FileImplPosix() {
   }
 }
 
-void FileImplPosix::openFile() {
-  const int flags = O_RDWR | O_APPEND | O_CREAT;
-  const int mode = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
-
-  fd_ = ::open(path_.c_str(), flags, mode);
+void FileImplPosix::openFile(const FlagSet in) {
+  auto flags = translateFlag(in);
+  fd_ = ::open(path_.c_str(), std::get<0>(flags), std::get<1>(flags));
```

Comment:
Move translateFlag into platform implementation.

Reply:
This line is hard to read. Can you make it easier to understand what is happening here?